src/main/kotlin/pl/edu/agh/server/domain/authentication/AuthenticationService.kt

```diff
@@ -28,13 +36,53 @@ class AuthenticationService(
             role = Role.USER,
         )
         userRepository.save(user)
-
-        return AuthenticationResponse(jwtService.generateToken(user))
+        val accessToken = jwtService.generateToken(user)
+        saveUserToken(user, accessToken)
+        return AuthenticationResponse(accessToken, jwtService.generateRefreshToken(user))
     }
 
     fun authenticate(request: AuthenticationRequest): AuthenticationResponse {
         authenticationManager.authenticate(UsernamePasswordAuthenticationToken(request.email, request.password))
         val user = userRepository.findByEmail(request.email).orElseThrow { throw Exception("User not found") }
-        return AuthenticationResponse(jwtService.generateToken(user))
+        revokeAllUserTokens(user.id!!)
```

Comment:
W tym momencie jak wywolasz `revokeAll` to ubijesz wszystkie pzostale sesje uzytkownika. Pytanie do klientki czy chcecmy wspierac mozliwosc korzystania z apki na kilku urzadzeniach